libnetdata/popen/popen.c

```diff
@@ -2,46 +2,66 @@
 
 #include "../libnetdata.h"
 
-/*
+static pthread_mutex_t myp_lock;
+static int myp_tracking = 0;
+
 struct mypopen {
     pid_t pid;
-    FILE *fp;
     struct mypopen *next;
     struct mypopen *prev;
 };
 
 static struct mypopen *mypopen_root = NULL;
 
-static void mypopen_add(FILE *fp, pid_t *pid) {
-    struct mypopen *mp = malloc(sizeof(struct mypopen));
-    if(!mp) {
-        fatal("Cannot allocate %zu bytes", sizeof(struct mypopen))
+static void myp_add(pid_t pid) {
+    struct mypopen *mp;
+
+    if (myp_tracking == 0)
+        return;
+
+    mp = mallocz(sizeof(struct mypopen));
+    if (mp == NULL) {
```

Comment:
Forgot to mention, `mallocz()` already does the `fatal()` inside so you can remove this `if` statement.